src/java.base/share/classes/java/util/TimeZone.java

```diff
@@ -395,9 +399,15 @@ public final String getDisplayName(boolean daylight, int style) {
      *                 {@code false} specifying a Standard Time name
      * @param style either {@link #LONG} or {@link #SHORT}
      * @param locale   the locale in which to supply the display name.
+     * @implSpec The default implementation throws a
+     * {@code NullPointerException} if {@code ID} is {@code null}.
+     * Additionally, an {@code IllegalArgumentException} will be thrown if
```

Comment:
I'd concatenate these with "or". Also, I think `IllegalArgumentException` has higher priority in the case of both being null/invalid.

Reply:
Swapped the priority, and improved the wording